File: 0383-ransom-note/0383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] count = new int[26];
        
        // Count characters in magazine
        for (char c : magazine.toCharArray()) {
            count[c - 'a']++;
        }
        
        // Check ransomNote
        for (char c : ransomNote.toCharArray()) {
            if (count[c - 'a']-- <= 0) {
                return false;
            }
        }
        
        return true;
    }
}

File: 0383-ransom-note/0383-ransom-note.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        vector<int> count(26, 0);
        
        // Count frequencies in magazine
        for (char c : magazine) {
            count[c - 'a']++;
        }
        
        // Check if ransomNote can be constructed
        for (char c : ransomNote) {
            if (count[c - 'a'] == 0) {
                return false;
            }
            count[c - 'a']--;
        }
        
        return true;
    }
};